### PR TITLE
Fix static_cast logic in hipify script for better coverage

### PR DESCRIPTION
**Pull request by jithunnair-amd**
Fix static_cast logic in hipify script for better coverage
Also remove obsolete code for output directory not existing

### CODE REVIEW

**Comment by iotamudelta** [comment]
Thanks! Did you check if unit tests are running?

**Comment by iotamudelta** [comment]
As per the review for #8812 w/ upstream: could you add to the documentation of the function an example for before vs after? Thanks

File: tools/amd_build/pyHIPIFY/hipify-python.py [hunk 1]
```diff
@@ -740,21 +740,20 @@ def add_static_casts(directory, extensions, KernelTemplateParams):
                             kernel_params = argument_strings[5:]
                             for arg_idx, arg in enumerate(kernel_params):
                                 if arg_idx in argument_types:
-                                    arg = kernel_params[arg_idx].strip()
                                     the_type = argument_types[arg_idx]
-                                    the_arg = arg.replace("\n", "").strip()
+                                    the_arg = arg.replace("\n", "").replace("\\", "").strip()
```
Comment (Jorghi12) [inline tools/amd_build/pyHIPIFY/hipify-python.py]: Why is replace("\\\\", "") added here?

Comment (jithunnair-amd) [reply tools/amd_build/pyHIPIFY/hipify-python.py]: There are instances where the kernel invocation is part of a preprocessor multiline macro. That's where the "\\" comes from (for the literal backslash used in a multiline macro).

**Comment by iotamudelta** [comment]
LGTM after the recent documentation additions. Thanks!

**Pull request merged**